spec/lib/mcb/commands/courses/list_spec.rb

```diff
@@ -0,0 +1,94 @@
+require 'mcb_helper'
+
+describe 'mcb providers list' do
+  def list(*arguments)
+    stderr = nil
+    output = with_stubbed_stdout(stdin: "", stderr: stderr) do
+      $mcb.run %W[courses list] + arguments
+    end
+    { stdout: output, stderr: stderr }
+  end
+
+  let(:current_cycle) { find_or_create(:recruitment_cycle, year: '2019') }
+  let(:additional_cycle) { find_or_create(:recruitment_cycle, year: '2020') }
+
+  context 'when recruitment cycle is unspecified' do
+    let(:provider1) { create(:provider, recruitment_cycle: current_cycle) }
+    let(:provider2) { create(:provider, recruitment_cycle: current_cycle) }
+
+    let(:course1) { create(:course, provider: provider1) }
+    let(:course2) { create(:course, provider: provider2) }
+
+    context 'when course is specified' do
+      it 'displays the provider information' do
+        course1
+        course2
+
+        command_output = list(course1.course_code)[:stdout]
+
+        expect(command_output).to include(course1.course_code)
+        expect(command_output).to include(course1.name)
+
+        expect(command_output).not_to include(course2.course_code)
+        expect(command_output).not_to include(course2.name)
+      end
+
+      it 'displays multiple specified courses' do
+        course1
+        course2
+
+        command_output = list(course1.course_code, course2.course_code)[:stdout]
+
+        expect(command_output).to include(course1.course_code)
+        expect(command_output).to include(course1.name)
+
+        expect(command_output).to include(course2.course_code)
+        expect(command_output).to include(course2.name)
+      end
+
+      it 'is case insensitive' do
+        course1
+        course2
+
+        command_output = list(course2.course_code.downcase)[:stdout]
+        expect(command_output).to include(course2.course_code)
+      end
+    end
+
+    context 'when provider is unspecified' do
+      it 'displays information about courses for the current recruitment cycle' do
+        course1
+        course2
+
+        command_output = list[:stdout]
+
+        expect(command_output).to include(course1.course_code)
+        expect(command_output).to include(course1.name)
+
+        expect(command_output).to include(course2.course_code)
+        expect(command_output).to include(course2.name)
+      end
+    end
+  end
+
+  context 'when recruitment cycle is specified' do
```

Comment:
How about a test that when recruitment-cycle is not specified, it defaults to the current one?